llmfoundry/callbacks/model_gauntlet_callback.py

```diff
@@ -90,18 +98,16 @@ def __init__(self,
                 assert weight is not None
                 benchmark['weighting'] = weight
 
-    def compute_averages(self, logger_data: Logger):
-
+    def compute_averages(self, logger_destination: InMemoryLogger):
         results = {}
         pat = re.compile(
             'metrics/(.*?)/(\d+)-shot(/.*?)?/InContextLearning(.*)'  # type: ignore
         )
         for key in self.logger_keys:
             match = pat.match(key)
-
-            # TODO(bmosaicml) This needs to be factored for this callback to work as a normal callback
-            # and therefore for the typing to be fixed
-            val = logger_data.data[key][0][1].item()  # type: ignore
+            if key not in logger_destination.data:
+                continue
+            val = logger_destination.data[key][-1][1].item()
```

Comment:
was the index change from `[0]` -> `[-1]` intentional here?

Reply:
yes, since logging events to the same key append to a list, in order to retrieve the most recent logging we need to use index -1

Reply:
`state.eval_metric_values` will contain the most recent eval metrics